src/vs/workbench/contrib/terminal/node/terminalProcess.ts

```diff
@@ -70,12 +70,22 @@ export class TerminalProcess implements ITerminalChildProcess, IDisposable {
 		};
 
 		try {
-			this._ptyProcess = pty.spawn(shellLaunchConfig.executable!, shellLaunchConfig.args || [], options);
-			this._processStartupComplete = new Promise<void>(c => {
-				this.onProcessIdReady((pid) => {
-					c();
+			const filePath = path.basename(shellLaunchConfig.executable!);
+			if (fs.existsSync(filePath)) {
```

Comment:
Unfortunately we can't use existsSync as that will block the whole thread and could cause issues particularly on slow HDDs. To do this properly I think we would need to change `_ptyProcess` to a `pty.IPty | undefined` and create it asynchronously and then fix any compile errors that come out of that. It's probably a good idea to do that anyway as `pty.spawn` is synchronous as well which we should fix eventually.

We already have `_processStartupComplete` which signals when the process is ready so hopefully there isn't too much to do here.

Reply:
Done and fixed the compile errors , The only thing i have not touched is changing pty.spawn to async functions , we might have to do that later as per your comments.